stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/sounds/Sounds.kt

```diff
@@ -16,16 +16,52 @@
 
 package io.getstream.video.android.core.sounds
 
+import android.content.Context
+import android.media.RingtoneManager
+import android.net.Uri
 import androidx.annotation.RawRes
+import io.getstream.log.taggedLogger
 import io.getstream.video.android.core.R
 
 /**
  * Contains all the sounds that the SDK uses.
  *
- * @param incomingCallSound Resource used as a ringtone for incoming calls.
- * @param outgoingCallSound Resource used as a ringing tone for outgoing calls.
+ * @param context Context to be used for retrieving the sounds.
+ * @param incomingCallSoundResId Resource to be used as a ringtone for incoming calls. Set to [DEVICE_INCOMING_RINGTONE] to use the device ringtone.
+ * @param outgoingCallSoundResId Resource to be used as a ringing tone for outgoing calls.
  */
-data class Sounds(
-    @RawRes val incomingCallSound: Int? = R.raw.call_incoming_sound,
-    @RawRes val outgoingCallSound: Int? = R.raw.call_outgoing_sound,
-)
+class Sounds(
+    private val context: Context,
+    @RawRes private val incomingCallSoundResId: Int? = DEVICE_INCOMING_RINGTONE,
+    @RawRes private val outgoingCallSoundResId: Int? = R.raw.call_outgoing_sound,
+) {
+    private val logger by taggedLogger("StreamVideo:Sounds")
+
+    internal val incomingCallSoundUri: Uri?
+        get() = if (incomingCallSoundResId == DEVICE_INCOMING_RINGTONE) {
+            RingtoneManager.getActualDefaultRingtoneUri(context, RingtoneManager.TYPE_RINGTONE) ?: run {
```

Comment:
> wrap this in safeCall or runCatching just in case we are failing the loading of the default ringtone.

The docs say that method returns `null` if the loading fails. In that case, we fall back to the default sound.

> Actually, why are we parsing this every time we get it?

That was because the ringtone can change since the last time the SDK was initialized.

> Since the context was used up we are not leaking it in the Sounds object

It uses the `context` that's already passed to the builder and which is stored anyway in `StreamVideoImpl`.

> This way we expose I think an unlimited number of possibilities with configuring the sounds, while having two easy default options. WDYT?

Overall looks like a good idea! We could use this opportunity and make it more flexible. 👍Maybe small changes like: the default should use the device ringtone etc. I'll start looking into it. ⚙️

Reply:
Refactored the code.